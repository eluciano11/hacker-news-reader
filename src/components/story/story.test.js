import React from "react";
import { render, cleanup } from "@testing-library/react";

import Story from "./story";

afterEach(cleanup);

describe("Story Component Tests", () => {
  it("Should render empty story", () => {
    const { asFragment } = render(<Story />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("Should render story with data", () => {
    const { asFragment } = render(
      <Story
        title="Title"
        author="Author"
        url="https://www.hackernews.com"
        date="August 1, 2019 1:00pm"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
