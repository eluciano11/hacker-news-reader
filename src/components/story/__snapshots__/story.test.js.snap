// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Component Tests Should render empty story 1`] = `
<DocumentFragment>
  <li
    class="story app__text--ellipsis"
  >
    <a
      class="app__link"
      href="/"
      rel="noopener noreferrer"
      target="_blank"
    />
    <p
      class="author"
    >
       | 
      <time
        class="author"
      />
    </p>
  </li>
</DocumentFragment>
`;

exports[`Story Component Tests Should render story with data 1`] = `
<DocumentFragment>
  <li
    class="story app__text--ellipsis"
  >
    <a
      class="app__link"
      href="https://www.hackernews.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Title
    </a>
    <p
      class="author"
    >
      Author | 
      <time
        class="author"
      >
        August 1, 2019 1:00pm
      </time>
    </p>
  </li>
</DocumentFragment>
`;
